src/shared/Core/Interop/Posix/PosixSessionManager.cs

```diff
@@ -1,17 +1,15 @@
-using System;
-
 namespace GitCredentialManager.Interop.Posix
 {
-    public class PosixSessionManager : SessionManager
+    public abstract class PosixSessionManager : SessionManager
     {
-        public PosixSessionManager()
+        protected PosixSessionManager(IEnvironment env, IFileSystem fs) : base(env, fs)
         {
             PlatformUtils.EnsurePosix();
         }
 
         // Check if we have an X11 or Wayland display environment available
         public override bool IsDesktopSession =>
-            !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("DISPLAY")) ||
-            !string.IsNullOrWhiteSpace(Environment.GetEnvironmentVariable("WAYLAND_DISPLAY"));
+            !string.IsNullOrWhiteSpace(System.Environment.GetEnvironmentVariable("DISPLAY")) ||
+            !string.IsNullOrWhiteSpace(System.Environment.GetEnvironmentVariable("WAYLAND_DISPLAY"));
```

Comment:
Why the addition of `System` here and removal of the `using`?

Reply:
Ahh.. good point. It's using `System.Environment` because we were using the BCL `Environment` class to read these variables initally.

Now we've added an `IEnvironment` instance to the `SessionManager`, there's a property named `Environment` (same name), so the namespace prefix is required to still reference the BCL static class.

But.. we should probably be using the `IEnvironment` here now that we have it.

Reply:
Pushed a new version, with a new commit at the front of the series that refactors the `IEnvironment` hierarchy a bit to remove a code smell.